### DISCUSSION TITLE
Allow multi-level import of config files

### DISCUSSION THREAD

**Discussion by hkleen** [INTI-CMNB/KiBot#218] [Ideas]
First of all, thanks for your great work, KiBot is really a great tool and makes KiCad so much more useful for us.

We have a number of different PCB projects and basically I'd like to generate similar outputs for all of them. Of course I could just copy over the configuration file and make the necessary adjustments on a per project basis, but this seems rather tedious and error prone. I tried the import mechanism in KiBot to come up with a more elegant solution, which kind of works, but I think being able to include configuration files inside another configuration file, i.e. recursive imports, would really make my life easier.

For example I would like to generate gerber files for each of my projects, but depending on project settings (number of layers, etc.), some projects need to be handled slightly different. If recursive imports where possible, I could write a configuration file called "gerber_base.yaml" where I configure the output files that are needed for every project (let's say the drill files and the report). I could then derive another configuration called "gerber_2layers.yaml" where I'd import "gerber_base.yaml" and add the outputs needed for a 2 layer board. Likewise I could have a "gerber_4layers.yaml" for 4 layer boards. In the projects configuration I would then include only the derived configuration file, e.g. "gerber_2layers.yaml"

Looking forward to hearing what you think about this idea

**Comment by set-soft** [comment]
The idea is good. I must evaluate the complexity.

Current code can't do it. This is because the initial idea was just import some outputs from a file and it started to get more and more complex ;-)

**Reply by set-soft** [reply]
Ok, I added a first implementation, and also added issue #220 
@hkleen let me know if the current implementation works for what you need.

**Reply by hkleen** [reply]
Now that was quick. I just tested the nested imports and as far as I can tell everything works as I had hoped. Great work!
